cli/src/commands/extensions/api.rs

```diff
@@ -465,123 +405,83 @@ fn run_sandboxed(op_state: Rc<RefCell<OpState>>, process: Process) -> Result<Pro
-            command.args(&args);
-            command.stdin(stdin);
-            command.stdout(Stdio::inherit());
-            command.stderr(Stdio::inherit());
-
-            // Wait for process to complete.
-            let status = command.status()?;
-
-            // Terminate child process.
-            if let Some(code) = status.code() {
-                process::exit(code);
-            } else if let Some(signal) = status.signal() {
-                // Attempt to propagate kill signals.
-                unsafe { libc::kill(process::id() as i32, signal) };
-
-                // Fall back to arbitrary error.
-                process::exit(113);
-            } else {
-                process::exit(0);
-            }
-        },
-        // Handle parent process.
-        child_pid => {
-            // Drop write side of pipe FDs, so child can write to it.
-            stdout_fds.child.take();
-            stderr_fds.child.take();
-
-            // Wait for the child to complete.
-            let mut exit_code = 0;
-            if unsafe { libc::waitpid(child_pid, (&mut exit_code) as *mut _, 0) } == -1 {
-                return Err(io::Error::last_os_error().into());
-            };
-
-            // Check process exit status.
-            let mut signal = None;
-            let mut code = None;
-            if libc::WIFEXITED(exit_code) {
-                code = Some(libc::WEXITSTATUS(exit_code));
-            } else if libc::WIFSIGNALED(exit_code) {
-                signal = Some(libc::WTERMSIG(exit_code));
-            } else if libc::WIFSTOPPED(exit_code) {
-                signal = Some(libc::WSTOPSIG(exit_code));
-            }
+    // Add sandbox subcommand argument.
+    let mut sandbox_args = Vec::with_capacity(args.len());
+    sandbox_args.push("sandbox".into());
 
-            // Read STDOUT/STDERR from pipe.
-            let mut stdout = String::new();
-            if let Some(mut stdout_fd) = stdout_fds.parent {
-                stdout_fd.read_to_string(&mut stdout)?;
-            }
-            let mut stderr = String::new();
-            if let Some(mut stderr_fd) = stderr_fds.parent {
-                stderr_fd.read_to_string(&mut stderr)?;
-            }
+    // Create CLI arguments for `phylum sandbox` permission exceptions.
+    add_permission_args(&mut sandbox_args, &resolved_permissions, strict)?;
 
-            Ok(ProcessOutput { stdout, stderr, success: code == Some(0), signal, code })
-        },
+    // Add sandboxed command arguments.
+    sandbox_args.push("--".into());
+    sandbox_args.push(cmd.into());
+    for arg in &args {
+        sandbox_args.push(arg.into());
     }
-}
 
-/// Lock down the current process.
-#[cfg(unix)]
-fn lock_process(exceptions: permissions::Permissions, strict: bool) -> Result<()> {
-    let home_dir = dirs::home_dir()?;
+    // Execute sandboxed command.
+    let output = Command::new(env::current_exe()?)
+        .args(sandbox_args.iter_mut().map(|arg| arg.to_mut()))
```

Comment:
The mutable references here feel wrong. There's got to be some way to make it work with something like this:

```suggestion
        .args(sandbox_args.into_iter().map(|arg| arg.as_ref()))
```

Although maybe it needs some hint to know that `.as_ref()` should return `&str` in that case.

Reply:
Can't use `into_iter` because then the reference doesn't last long enough.

Can't use `as_ref` for similar reasons, `.borrow()` is the only way we could convert here afaik but that requires massive type annotations so it's not worth it. Afaik `to_mut` is the most concise way to transform a `Cow<'_, T>` into `&T` in this scenario (I've tried a couple things actually).